### PR TITLE
HAI-1842 Add endpoint for unlinking applications from Allu in dev and test

### PR DESCRIPTION
**Pull request by corvidian**
# Description

Allu removes all of its data in the test environment when it's updated. To mitigate problems this causes to Haitaton dev and test environments, add an endpoint that removes the linking between Haitaton applications and Allu. The links are non-functional after an Allu update anyway.

Make the endpoint callable directly from Swagger UI, so it calling it doesn't require a developer.

The endpoint is only visible and accessible from specific environments, namely dev and test environments. This is accomplished with environment variables.

## Type of change

- [ ] Bug fix 
- [ ] New feature 
- [X] Other

# Instructions for testing
Create an application and then call the endpoint to unlink it.

# Checklist:

- [X] I have written new tests (if applicable)
- [X] I have ran the tests myself (if applicable)
- [ ] I have made necessary changes to the documentation, link to confluence
 or other location: TODO

### CODE REVIEW

File: services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/testdata/TestDataServiceITest.kt [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+    @Autowired private lateinit var testDataService: TestDataService
+    @Autowired private lateinit var applicationRepository: ApplicationRepository
+    @Autowired private lateinit var alluDataFactory: AlluDataFactory
+    @Autowired private lateinit var hankeFactory: HankeFactory
+
+    @Nested
+    inner class UnlinkApplicationsFromAllu {
+        @Test
+        fun `Without applications does nothing`() {
```
Comment (pitkni) [inline services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/testdata/TestDataServiceITest.kt:40]: Does this actually verify that it does not do anything?

Comment (corvidian) [reply services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/testdata/TestDataServiceITest.kt]: The important thing is that it doesn't throw an exception with an empty list. I renamed the test.

File: services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/security/AccessRules.kt [hunk 2]
```diff
@@ -26,6 +26,11 @@ object AccessRules {
                 "/v3/api-docs/**",
             )
             .permitAll()
+            .mvcMatchers(HttpMethod.POST, "/testdata/unlink-applications")
```
Comment (pitkni) [inline services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/security/AccessRules.kt:29]: Would it be better to require an access token and log the user doing this operation? I don't have a clear opinion myself so I'm just pondering.

Comment (corvidian) [reply services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/security/AccessRules.kt:29]: Requiring an access token would in practice limit the use to developers.

**Review by pitkni — APPROVED** [review]

**Pull request merged**